src/Symfony/Component/Translation/MessageCatalogue.php

```diff
@@ -157,8 +166,12 @@ public function add(array $messages, string $domain = 'messages')
     {
         $altDomain = str_ends_with($domain, self::INTL_DOMAIN_SUFFIX) ? substr($domain, 0, -\strlen(self::INTL_DOMAIN_SUFFIX)) : $domain.self::INTL_DOMAIN_SUFFIX;
         foreach ($messages as $id => $message) {
-            unset($this->messages[$altDomain][$id]);
-            $this->messages[$domain][$id] = $message;
+            if (null === $message) {
+                unset($this->messages[$domain][$id]);
+            } else {
+                unset($this->messages[$altDomain][$id]);
+                $this->messages[$domain][$id] = (string) $message;
```

Comment:
But the `(string) $message` cast will throw an exception if for objects/resources that can't be converted into strings. Maybe it's what we want (to fail loudly so you can catch these errors) but I just wanted to mention it. Thanks.